lib/core/services/location_service.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:ata/core/models/failure.dart';
+import 'package:ata/core/models/location.dart';
+import 'package:ata/core/services/office_service.dart';
+import 'package:dartz/dartz.dart';
+import 'package:geolocator/geolocator.dart';
+
+class LocationService {
+  final OfficeService _officeService;
+
+  LocationService(OfficeService officeService) : _officeService = officeService;
+
+  Future<Either<Failure, bool>> checkDistanceLocation(Location deviceLocation, Location officeLocation, double authRange) async {
+    final double startLatitude = deviceLocation.lat;
+    final double startLongitude = deviceLocation.lng;
+    final double endLatitude = officeLocation.lat;
+    final double endLongitude = officeLocation.lng;
+    final double distance = await Geolocator().distanceBetween(startLatitude, startLongitude, endLatitude, endLongitude);
+    return Right((distance / 1000 <= authRange) ? true : false);
```

Comment:
distanceBetween is already in meters, authRange also in meters, why need to /1000